google-pubsub/1.0/modules/ROOT/pages/google-pubsub-connector-reference.adoc

```diff
@@ -0,0 +1,641 @@
+
+Note that this operation affects only messages retained in the subscription (configured by the combination of *messageRetentionDuration* and *retainAckedMessages*), for example, if the time corresponds to a point before the message retention window (or to a point before the system's notion of the subscription creation time), only retained messages are marked as unacknowledged, and already-expunged messages will not be restored. A timestamp is in RFC3339 UTC Zulu format, with nanosecond resolution and up to nine fractional digits, such as 2014-10-02T15:01:23Z and 2014-10-02T15:01:23.045123456Z. |  |
+| Reconnection Strategy a| * <<Reconnect>>
+* <<ReconnectForever>> |  Retry strategy in case of connectivity errors. |  |
+|===
+
+
+=== For Configurations
+
+* <<Config>>
+
+=== Throws
+
+* PUBSUB:BAD_GATEWAY
+* PUBSUB:CANCELLED
+* PUBSUB:CONNECTIVITY
+* PUBSUB:DEADLINE_EXCEEDED
+* PUBSUB:FAILED_PRECONDITION
+* PUBSUB:INVALID_ARGUMENT
+* PUBSUB:NOT_FOUND
+* PUBSUB:PERMISSION_DENIED
+* PUBSUB:RETRY_EXHAUSTED
+* PUBSUB:UNAVAILABLE
+
+
+== Sources
+
+[[MessageListener]]
+== On message listener
+`<pubsub:message-listener>`
+
+
+Runnable message receiver, using Pub/Sub Subscriber.
+
+
+=== Parameters
+
+[%header%autowidth.spread]
+|===
+| Name | Type | Description | Default Value | Required
+| Configuration | String | Name of the configuration to use. | | x
+| Consumer count a| Number |  Provides a specified amount of executor service for processing messages. |  5 |
+| Config Ref a| ConfigurationProvider |  Name of the configuration used to execute this component. |  | x
+| Primary Node Only a| Boolean |  Determines whether to execute this source on only the primary node when running Mule instances in a cluster. |  |
+| Streaming Strategy a| * <<RepeatableInMemoryStream>>
+* <<RepeatableFileStoreStream>>
+* non-repeatable-stream |  Configures how Mule processes streams. The default is to use repeatable streams. |  |
+| Redelivery Policy a| <<RedeliveryPolicy>> |  Defines a policy for processing the redelivery of the same message. |  |
+| Project ID a| String |  Globally unique identifier for your project. |  |
+| Subscription Name a| String |  Project-wide unique identifier for the subscription. |  |
+| Limit Exceeded Behavior a| Enumeration, one of:
+
+** THROW_EXCEPTION
+** BLOCK
+** IGNORE |  If LimitExceededBehavior is not set to `IGNORE`, the flow control enables you to have control over the behavior of receiving messages by specifying the limits for *maxOutstandingRequestSize* and *maxOutstandingElementCount*. If the memory size or the amount of outstanding elements (messages where no Ack() or Nack() was performed) exceeds specified limits, the receiver won't receive more messages when LimitExceedBehavior is set to `BLOCK`, or will throw a FlowControlException when LimitExceedBehavior is set to `THROW_EXCEPTION`. |  BLOCK |
```

Comment:
See if you can rewrite this using a bullet for each possible value. I'll look at it again afterward.

Enables you to control the behavior of receiving messages:

* Throw_Exception...
* BLOCK...
* IGNORE...